lm_eval/tasks/mlqa.py

```diff
@@ -0,0 +1,287 @@
+from . common import HFTask
+from math import exp
+from functools import partial
+import sys, unicodedata, string
+import re
+from collections import Counter
+from lm_eval.base import rf
+
+PUNCT = {chr(i) for i in range(sys.maxunicode) if unicodedata.category(chr(i)).startswith('P')}.union(
+    string.punctuation)
+WHITESPACE_LANGS = ['en', 'es', 'hi', 'vi', 'de', 'ar']
+MIXED_SEGMENTATION_LANGS = ['zh']
+ALL_MLQA_SETS_TRANSLATE = ['mlqa-translate-train.ar', 'mlqa-translate-train.de', 'mlqa-translate-train.vi',
+                           'mlqa-translate-train.zh', 'mlqa-translate-train.es', 'mlqa-translate-train.hi',
+                           'mlqa-translate-test.ar', 'mlqa-translate-test.de', 'mlqa-translate-test.vi',
+                           'mlqa-translate-test.zh',
+                           'mlqa-translate-test.es', 'mlqa-translate-test.hi']
+
+ALL_MLQA_SETS_QA = ['mlqa.ar.ar', 'mlqa.ar.de', 'mlqa.ar.vi',
+                    'mlqa.ar.zh', 'mlqa.ar.en', 'mlqa.ar.es', 'mlqa.ar.hi', 'mlqa.de.ar', 'mlqa.de.de', 'mlqa.de.vi',
+                    'mlqa.de.zh', 'mlqa.de.en', 'mlqa.de.es', 'mlqa.de.hi', 'mlqa.vi.ar', 'mlqa.vi.de', 'mlqa.vi.vi',
+                    'mlqa.vi.zh', 'mlqa.vi.en', 'mlqa.vi.es', 'mlqa.vi.hi', 'mlqa.zh.ar', 'mlqa.zh.de', 'mlqa.zh.vi',
+                    'mlqa.zh.zh', 'mlqa.zh.en', 'mlqa.zh.es', 'mlqa.zh.hi', 'mlqa.en.ar', 'mlqa.en.de', 'mlqa.en.vi',
+                    'mlqa.en.zh', 'mlqa.en.en', 'mlqa.en.es', 'mlqa.en.hi', 'mlqa.es.ar', 'mlqa.es.de', 'mlqa.es.vi',
+                    'mlqa.es.zh', 'mlqa.es.en', 'mlqa.es.es', 'mlqa.es.hi', 'mlqa.hi.ar', 'mlqa.hi.de', 'mlqa.hi.vi',
+                    'mlqa.hi.zh', 'mlqa.hi.en', 'mlqa.hi.es', 'mlqa.hi.hi']
+
+ALL_MLQA_SETS = ALL_MLQA_SETS_QA + ALL_MLQA_SETS_TRANSLATE
+
+BG_Q_A_STRINGS = {
+    "en": {"bg": "Background:", "q": "Question:", "a": "Answer:"},
+    "ar": {"bg": ":معرفتي", "q": ":سؤال", "a": ":إجابه"},
+    "de": {"bg": "Hintergrund:", "q": "Frage:", "a": "Antwort:"},
+    "zh": {"bg": "背景:", "q": "問題:", "a": "答案:"},
+    "vi": {"bg": "lý lịch:", "q": "câu hỏi:", "a": "câu trả lời:"},
+    "es": {"bg": "antecedentes:", "q": "pregunta:", "a": "respuesta:"},
+    "hi": {"bg": "Ιστορικό:", "q": "ερώτηση:", "a": "απάντηση:"},
```

Comment:
Why is the thing for `hi` (Hindi) in Greek here?

Reply:
ah, good catch. bad copy and pasting